test/suite-functional/api/functional.link.spec.php

```diff
@@ -161,6 +161,31 @@
 
         $strandA->terminate();
     });
+
+    rit('can link to self', function () {
```

Comment:
Yeah ok, it was hard to word because i couldn't test that it did nothing, the mocks were interacted with. I'll go with that second option "does not fail if linked to itself".